components/timestamp/index.test.tsx

```diff
@@ -0,0 +1,142 @@
+                category: 'display_settings',
+                name: 'use_military_time',
+                user_id: currentUserId,
+                value: 'true',
+            } as PreferenceType;
+
+            const props = mapStateToProps(testState, {});
+            expect(props.hourCycle).toBe('h23');
+        });
+
+        test('hourCycle should have the value of prop.hourCycle when given', () => {
+            const testState = {...initialState};
+            testState.entities.preferences.myPreferences['display_settings--use_military_time'] = {
+                category: 'display_settings',
+                name: 'use_military_time',
+                user_id: currentUserId,
+                value: 'true',
+            } as PreferenceType;
+
+            const props = mapStateToProps(testState, {hourCycle: 'h24'});
+            expect(props.hourCycle).toBe('h24');
+        });
+    });
+
+    describe('timeZone', () => {
+        test('timeZone should be the user TZ when the prop was not set', () => {
+            const testState = {...initialState};
+            testState.entities.users.profiles[currentUserId].timezone = {
+                useAutomaticTimezone: false,
+                manualTimezone: 'Europe/Paris',
+            } as UserTimezone;
+
+            const props = mapStateToProps(testState, {});
+            expect(props.timeZone).toBe('Europe/Paris');
+        });
+
+        test('timeZone should be the value of prop.timeZone when given', () => {
+            const testState = {...initialState};
+            testState.entities.users.profiles[currentUserId].timezone = {
+                useAutomaticTimezone: false,
+                manualTimezone: 'Europe/Paris',
+            } as UserTimezone;
+
+            const props = mapStateToProps(testState, {timeZone: 'America/Phoenix'});
+            expect(props.timeZone).toBe('America/Phoenix');
+        });
+
+        test('timeZone should be the value of prop.timeZone when given, even when timezone are disabled', () => {
+            const testState = {...initialState};
+            testState.entities.general.config.ExperimentalTimezone = 'false';
+
+            const props = mapStateToProps(testState, {timeZone: 'America/Chicago'});
+            expect(props.timeZone).toBe('America/Chicago');
+        });
+    });
+
+    describe('hour12', () => {
+        test('hour12 should be false when using military time', () => {
+            const testState = {...initialState};
+            testState.entities.preferences.myPreferences['display_settings--use_military_time'] = {
+                category: 'display_settings',
+                name: 'use_military_time',
+                user_id: currentUserId,
+                value: 'true',
+            } as PreferenceType;
+
+            const props = doMapStateToProps(testState, {}, false);
+            expect(props.hour12).toBe(false);
+        });
```

Comment:
```suggestion
        test('hour12 should be false when using military time and hourCycle unsupported', () => {
            supportsHourCycleSpy.mockReturnValueOnce(false);
            const testState = {...initialState};
            testState.entities.preferences.myPreferences['display_settings--use_military_time'] = {
                category: 'display_settings',
                name: 'use_military_time',
                user_id: currentUserId,
                value: 'true',
            } as PreferenceType;

            const props = doMapStateToProps(testState, {});
            expect(props.hour12).toBe(false);
        });
```

(also for the other 2 tests in `hour12`)